src/gdbstub_impl/ext/reverse_exec.rs

```diff
@@ -0,0 +1,106 @@
+use super::prelude::*;
+use crate::protocol::{
+    commands::ext::{ReverseCont, ReverseStep},
+    SpecificIdKind,
+};
+
+use crate::target::ext::base::BaseOps;
+
+impl<T: Target, C: Connection> GdbStubImpl<T, C> {
+    pub(crate) fn handle_reverse_cont(
+        &mut self,
+        res: &mut ResponseWriter<C>,
+        target: &mut T,
+        command: ReverseCont,
+    ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
+        let handler_status = match command {
+            ReverseCont::bc(_) => {
+                // FIXME: This block is duplicated from the vCont code.
+                let mut err = Ok(());
+                let mut check_gdb_interrupt = || match res.as_conn().peek() {
+                    Ok(Some(0x03)) => true, // 0x03 is the interrupt byte
+                    Ok(Some(_)) => false,   // it's nothing that can't wait...
+                    Ok(None) => false,
+                    Err(e) => {
+                        err = Err(Error::ConnectionRead(e));
+                        true // break ASAP if a connection error occurred
+                    }
+                };
+
+                let stop_reason = match target.base_ops() {
+                    BaseOps::MultiThread(ops) => {
+                        if let Some(ops) = ops.support_reverse_cont() {
+                            ops.reverse_cont(&mut check_gdb_interrupt)
+                                .map_err(Error::TargetError)?
+                        } else {
+                            return Err(Error::PacketUnexpected);
+                        }
+                    }
+
+                    BaseOps::SingleThread(_ops) => {
+                        return Err(Error::PacketUnexpected);
+                    }
+                };
+
+                err?;
+
+                // FIXME: Unnecessary unwrap
```

Comment:
yeah... unwraps are a strict no-no in the `gdbstub` codebase. Unless you've double checked the generated asm and confirmed that the panic branch has been optimized out, this needs to be handled without `panic!`ing

The "proper" fix in the future would be to introduce some similar looping logic, but for now, it should be fine to simply `.ok_or(Error::PacketUnexpected)`. It's not _strictly_ correct, but it'll do for now.

Please add a `// FIXME: properly handle None case` comment as well.